cmds/find/find.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Find finds files. It is similar to the Unix command. It uses REs, not globs,
+// for matching.
+//
+// OPTIONS:
+//     -d: enable debugging in the find package
+//     -mode integer-arg: match against mode, e.g. -mode 0755
+//     -type: match against a file type, e.g. -type f will match files
+//     -name: glob to match against file
+//     -l: long listing. It's not very good, yet, but it's useful enough.
+//     -root: where to start the search, default /tmp
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/find"
+)
+
+var (
+	perm      = flag.Int("mode", -1, "Permissions")
+	fileType  = flag.String("type", "", "File type")
+	name      = flag.String("name", "", "glob for name")
+	root      = flag.String("root", "/tmp", "path for root")
+	long      = flag.Bool("l", false, "long listing")
+	debug     = flag.Bool("d", false, "Enable debugging in the find package")
+	fileTypes = map[string]os.FileMode{
+		"f":         0,
+		"file":      0,
+		"d":         os.ModeDir,
+		"directory": os.ModeDir,
+	}
+)
+
+func usage() {
+	flag.Usage()
+	os.Exit(1)
+}
+
+func main() {
+	flag.Parse()
+	if len(flag.Args()) > 0 {
+		usage()
+	}
+	var mask, mode os.FileMode
+	if *perm != -1 {
+		mask = os.ModePerm
+		mode = os.FileMode(*perm)
+	}
+	if *fileType != "" {
+		intType, ok := fileTypes[*fileType]
+		if !ok {
+			var keys []string
+			for key, _ := range fileTypes {
```

Comment:
Drop the `, _`, use `for key := range fileTypes` instead.

Reply:
done